packages/toggle/src/Toggle.tsx

```diff
@@ -389,122 +384,94 @@ const getStatefulStyles = (states: StateForStyle) => ({
   container: getContainerStyles(states),
 });
 
-interface ToggleProps {
-  size: Size;
-  variant: Variant;
-  checked?: boolean;
-  disabled: boolean;
-  className?: string;
-  onChange: React.ChangeEventHandler<HTMLInputElement>;
-  id?: string;
-}
-
-interface ToggleState {
-  checked: boolean;
+interface BaseToggleProps {
+  size?: Size;
+  darkMode?: boolean;
 }
 
-export default class Toggle extends PureComponent<
-  ToggleProps &
-    Omit<React.InputHTMLAttributes<HTMLInputElement>, keyof ToggleProps>,
-  ToggleState
-> {
-  static displayName = 'Toggle';
-
-  static propTypes = {
-    size: PropTypes.oneOf(['default', 'small', 'xsmall']),
-    variant: PropTypes.oneOf(['default', 'dark']),
-    checked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    className: PropTypes.string,
-    onChange: PropTypes.func,
-    name: PropTypes.string,
-    id: PropTypes.string,
-  };
-
-  static defaultProps = {
-    size: Size.Default,
-    variant: Variant.Default,
-    disabled: false,
-    className: '',
-    onChange: () => {},
-  };
-
-  state: ToggleState = { checked: false };
-
-  private static idAllocator = IdAllocator.create('toggle');
-  private _defaultCheckboxId?: string;
-
-  private get defaultCheckboxId() {
-    if (!this._defaultCheckboxId) {
-      this._defaultCheckboxId = `toggle-${Toggle.idAllocator.generate()}`;
+type ToggleProps = BaseToggleProps &
+  Omit<React.InputHTMLAttributes<HTMLInputElement>, keyof BaseToggleProps>;
+
+const idAllocator = IdAllocator.create('toggle');
+
+function Toggle({
+  size = Size.Default,
+  darkMode = false,
+  disabled = false,
+  className,
+  onChange: onChangeProp = () => {},
+  checked: checkedProp,
+  name,
+  id,
+  ...rest
+}: ToggleProps) {
+  const [checked, setChecked] = useState(false);
+  const checkboxId = id || `toggle-${idAllocator.generate()}`;
```

Comment:
I know the code was already like this, but this could actually be changed since `IdAllocator` already adds a prefix.
```suggestion
  const checkboxId = id || idAllocator.generate();
```

Reply:
also, this will generate a new ID on every render. I think there are other function components that demonstrate how to maintain a stable ID with `IdAllocator`

Reply:
👍